corehq/apps/users/models.py

```diff
@@ -1764,17 +1764,21 @@ def get_language_code(self):
     def location(self):
         from corehq.apps.locations.models import Location
         if self.location_id:
-            return Location.get(self.location_id)
-        else:
-            return None
+            try:
+                return Location.get(self.location_id)
+            except ResourceNotFound:
+                self.unset_location()
```

Comment:
I generally think we should try to avoid doing things that unexpectedly write to the database in properties. could this be changed to just raise a `LocationNotFoundException`, and then have another function called `get_or_unset_location` which you call when you want to automatically unset it? i realize that's more annoying to implement, but it does require the caller to explicitly acknowledge that they want to do work when the location id is bad, which I think ultimately is a good thing.